### PR TITLE
Adopt a shared AdminClient across selected CC components

### LINKED ISSUE #1340: Adopt a shared AdminClient across selected CC components
Several components of Cruise Control instantiate their own `AdminClient` for infrequent API accesses. In particular, these clients are created for (1) each request handled by `kafka_cluster_state` endpoint to make a single call to `describeLogDirs`, (2) `MaintenanceEventTopicReader` for a once in a lifetime topic creation / update, (3) `DiskFailureDetector` to `describeLogDirs` and `TopicReplicationFactorAnomalyFinder` to `describeConfigs` of new topics once in every corresponding anomaly detection interval.

Kafka's Admin Client is thread-safe (see https://stackoverflow.com/a/61738065). Hence, such expensive AdminClient instantiations can be eliminated by sharing a single instance.

### PR DESCRIPTION
**Pull request by efeg**
This PR resolves #1340.

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/KafkaClusterState.java [hunk 1]
```diff
@@ -27,18 +28,18 @@
   public static final String KAFKA_PARTITION_STATE = "KafkaPartitionState";
   protected final Map<String, Properties> _allTopicConfigs;
   protected final Properties _clusterConfigs;
-  protected final Map<String, Object> _adminClientConfigs;
+  protected final AdminClient _adminClient;
```
Comment (smccauliff) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/KafkaClusterState.java:31]: Is the AdminClient thread-safe?  I couldn't find a yes or no answer in the javadocs for KafkaAdminClient class or Admin interface searching for "thread" or "safe".

Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/KafkaClusterState.java:31]: The linked issue, i.e. https://github.com/linkedin/cruise-control/issues/1340, seems to be the only source verifying the thread safety of AdminClients -- i.e. confirmed by @cmccabe at https://stackoverflow.com/a/61738065.

Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/KafkaClusterState.java:31]: `Update`: I created [KAFKA-10583](https://issues.apache.org/jira/browse/KAFKA-10583) along with the [corresponding PR](https://github.com/apache/kafka/pull/9397) involving the JavaDoc update.

Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/KafkaClusterState.java:31]: @smccauliff Can we take another look at this PR? -- [KAFKA-10583](https://issues.apache.org/jira/browse/KAFKA-10583) has been addressed.

**Review by smccauliff — APPROVED** [review]

**Pull request merged**